modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/rebalancing/RebalanceMetricsTest.java

```diff
@@ -0,0 +1,111 @@
+
+import org.apache.ignite.cache.CacheMetrics;
+import org.apache.ignite.cluster.ClusterState;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.cache.CacheMetricsImpl;
+import org.apache.ignite.internal.processors.cache.GridCacheContext;
+import org.apache.ignite.internal.util.typedef.internal.CU;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.jetbrains.annotations.Nullable;
+import org.junit.Test;
+
+/**
+ * Class for testing rebalance metrics.
+ */
+public class RebalanceMetricsTest extends GridCommonAbstractTest {
+    /** Cache configurations. */
+    @Nullable private CacheConfiguration<?, ?>[] cacheConfigs;
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName).setCacheConfiguration(cacheConfigs);
+    }
+
+    /**
+     * Testing the correctness of metric {@link CacheMetrics#getRebalancedKeys()}.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testReceivedKeys() throws Exception {
+        cacheConfigs = new CacheConfiguration[] {
+            new CacheConfiguration<>(DEFAULT_CACHE_NAME + "0")
+                .setGroupName(DEFAULT_CACHE_NAME)
+                .setStatisticsEnabled(true),
+            new CacheConfiguration<>(DEFAULT_CACHE_NAME + "1")
+                .setGroupName(DEFAULT_CACHE_NAME)
+                .setStatisticsEnabled(true),
+            new CacheConfiguration<>(DEFAULT_CACHE_NAME + "2")
+                .setGroupName(DEFAULT_CACHE_NAME)
+                .setStatisticsEnabled(false),
+        };
+
+        IgniteEx n0 = startGrid(0);
+
+        n0.cluster().state(ClusterState.ACTIVE);
+        awaitPartitionMapExchange();
+
+        for (CacheConfiguration<?, ?> cacheCfg : cacheConfigs) {
+            for (int i = 0; i < 100; i++)
+                n0.cache(cacheCfg.getName()).put(i + CU.cacheId(cacheCfg.getName()), new byte[64]);
+        }
+
+        IgniteEx n1 = startGrid(1);
+        awaitPartitionMapExchange();
+
+        for (CacheConfiguration<?, ?> cacheCfg : cacheConfigs) {
+            GridCacheContext<?, ?> cacheCtx0 = n0.context().cache().cache(cacheCfg.getName()).context();
+            GridCacheContext<?, ?> cacheCtx1 = n1.context().cache().cache(cacheCfg.getName()).context();
+
+            assertEquals(cacheCfg.getName(), cacheCfg.isStatisticsEnabled(), cacheCtx0.statisticsEnabled());
```

Comment:
I would like a human-readable message here and below.
This one does not seem good enough:
```
java.lang.AssertionError: default0 
Expected :false
Actual   :true
```

Reply:
I don't think it will help much.